components/ws-daemon/pkg/daemon/daemon.go

```diff
@@ -52,12 +53,22 @@ func NewDaemon(config Config, reg prometheus.Registerer) (*Daemon, error) {
 	if err != nil {
 		return nil, err
 	}
-	dsptch, err := dispatch.NewDispatch(containerRuntime, clientset, config.Runtime.KubernetesNamespace, nodename,
+
+	listener := []dispatch.Listener{
 		cpulimit.NewDispatchListener(&config.Resources, reg),
-		CacheReclaim(config.Resources.CGroupBasePath),
 		cgCustomizer,
 		markUnmountFallback,
-	)
+	}
+
+	unified, err := cgroups.IsUnifiedCgroupSetup()
+	if err != nil {
+		return nil, xerrors.Errorf("could not determine cgroup setup: %w", err)
+	}
+	if unified {
```

Comment:
Shouldn't this be the other way around? That is add cache reclaim if we are on a system with a legacy (v1) cgroup setup?

Reply:
you are right! Thanks.